src/forms/formHelpers.js

```diff
@@ -443,50 +514,54 @@ class CashFlowInput extends Component {
 */
 const CashFlowRow = function ({ generic, timeState, setClientProperty, children, labelInfo, type, time }) {
 
+  var updateClient = function ( evnt, value, interval ) {
+    var monthly = toMonthlyAmount[ interval ]( evnt, value ),
+        obj     = { name: generic, value: monthly };
+    setClientProperty( evnt, obj );
+  };
```

Comment:
You could get around passing interval to the `<ManagedInput>` if you wanted it to be extra general.

```js
const updateClient = (interval) => (
  (evnt, value) => {
    const monthly = toMonthlyAmount[ interval ]( evnt, value );
    const obj = { name: generic, value: monthly };
    setClientProperty( evnt, obj );
  }
);
```
Then when you are defining your inputs:

```js
<ManagedInput {...props} store={updateClient('weekly')} />
```

Reply:
It's an interesting solution and I'll have to think about it. I find that kind of function nesting more complex to hold in my mind when coding, so I tend not to use it where possible. There's at least one place in the code base where do that and I'm not super happy with it.